### PR TITLE
Payment step transaction verification of the data

### PR DESCRIPTION
**Pull request by PaulIRVOAS**

### CODE REVIEW

**Comment by PaulIRVOAS** [comment]
Requested by @MeloLawson ->

- TransactionController -> please save a Transaction object in DB + set transaction status before pushing in DB.
- PaymentMethodService -> take a look on the wanted account object, it is recovered ?
- Mapper

File: Server/bank/transaction/src/main/java/com/cashmanager/server/transaction/controller/TransactionController.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    @PostMapping("/transaction")
+    public void checkTransaction(@RequestBody TransactionHolder transactionHolder) {
+
+        PaymentMethodDto paymentMethodDto = transactionHolder.getPaymentMethodDto();
+        TransactionDto transactionDto = TransactionHolder.setupTransactionDto(transactionHolder);
+
+        Optional<PaymentMethod> paymentMethod = paymentMethodService.checkPaymentMethodViability(paymentMethodDto, transactionDto);
+        if (paymentMethod.isPresent()) {
+            if (accountService.checkAccountValidity(accountRepository.findById(paymentMethod.get().getAccount().getId()), transactionDto)) {
```
Comment (MeloLawson) [inline Server/bank/transaction/src/main/java/com/cashmanager/server/transaction/controller/TransactionController.java:49]: for this part, what i meant was to put the idAccount in the checkAccountValidity(), and in the method you use the accountRepository.findById(idAccount).
 ```suggestion
            if (accountService.checkAccountValidity(paymentMethod.get().getAccount().getId(), transactionDto)) {
```
To simplify this, you can even change the type return of the checkPaymentMethodViability to Optional&lt;UUID&gt; to return directly the account id and don't use the entity PaymentMethod in the controller.

**Review by MeloLawson — CHANGES_REQUESTED** [review]
Just one suggestion about in the controller/paymentMethodService

File: Server/bank/transaction/src/main/java/com/cashmanager/server/transaction/controller/TransactionController.java (hunk 1 above)
Comment (PaulIRVOAS) [reply Server/bank/transaction/src/main/java/com/cashmanager/server/transaction/controller/TransactionController.java]: Thx for the review, I have applied your first suggestion.
Not the second, it's less understandable for the code on my view.

**Review by MeloLawson — APPROVED** [review]

**Pull request merged**